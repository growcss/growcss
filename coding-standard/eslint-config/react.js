module.exports ={
  extends: [
    'prettier/react',
    'plugin:jest/recommended'
  ],
  plugins: [
    'react',
    'jsx-a11y',
    // 'security', @todo check it on the next version
  ],
  rules: {
    ...{
      'react/sort-comp': 'off',
      'react/jsx-filename-extension': 'off',
      'react/require-default-props': 'off',
    },
    ...require('./rules/prettier-conflicts-react'),
    ...require('./rules/jest'),
  },
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },
};
