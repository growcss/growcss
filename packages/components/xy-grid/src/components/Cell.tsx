import * as React from 'react';
import classNames from 'classnames';
import { Gutters as DefaultGutters, GuttersProps } from './Gutters';
import { CellElement } from '../styled/CellElement';

export interface CellProps extends React.HTMLAttributes<HTMLDivElement>{
  children: React.ReactNode,
  gridColumns?: number,
  vertical?: boolean,
  gutterSizes?: GuttersProps,
  cellType?: string,
  gutterType?: string,
  align?: string,
}

export interface OffsetProps {
  smallOffset?: number,
  mediumOffset?: number,
  largeOffset?: number,
  xlargeOffset?: number,
  xxlargeOffset?: number,
}

export interface CellBreakpointsProps {
  small?: number | string,
  medium?: number | string,
  large?: number | string,
  xlarge?: number | string,
  xxlarge?: number | string,
}

export default class Cell extends React.Component<CellProps & CellBreakpointsProps & OffsetProps> {
  static defaultProps = {
    gridColumns: 12,
    gutterSizes: DefaultGutters,
    vertical: false,
  };

  render() {
    const {
      children,
      gridColumns,
      cellType,
      gutterType,
      gutterSizes,
      vertical,
      align,
      ...other
    } = this.props;
    const className = classNames('gc-cell');

    return (
      <CellElement
        className={className}
        gridColumns={gridColumns!}
        cellType={cellType}
        gutterType={gutterType}
        gutterSizes={gutterSizes!}
        vertical={vertical!}
        align={align}
        {...other}
      >
        {children}
      </CellElement>
    );
  }
}
