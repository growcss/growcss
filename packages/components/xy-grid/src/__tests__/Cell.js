import * as React from 'react';
import Cell from '../components/Cell';
import 'jest-styled-components';

test('cell base layout', () => {
  const wrapper = shallow(<Cell>test</Cell>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '0 0 auto');
  expect(wrapper).toHaveStyleRule('width', '100%');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
});

test('cell auto layout with padding gutter', () => {
  const wrapper = shallow(
    <Cell cellType="auto" gutterType="padding">
      test
    </Cell>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '1 1 0px');
  expect(wrapper).toHaveStyleRule('width', 'auto');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
  expect(wrapper).toHaveStyleRule('box-sizing', 'border-box');
});

test('cell auto layout with margin gutter', () => {
  const wrapper = shallow(
    <Cell cellType="auto" gutterType="margin">
      test
    </Cell>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '1 1 0px');
  expect(wrapper).toHaveStyleRule('width', 'auto');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
});

test('cell shrink layout', () => {
  const wrapper = shallow(<Cell cellType="shrink">test</Cell>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '0 0 auto');
  expect(wrapper).toHaveStyleRule('width', 'auto');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
});

test('cell with 6 small layout', () => {
  const wrapper = shallow(<Cell small={6}>test</Cell>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '0 0 auto');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
  expect(wrapper).toHaveStyleRule('width', '50%');
  expect(wrapper).toHaveStyleRule('flex-basis', 'auto');
});

test('cell align', () => {
  const wrapper = shallow(<Cell align="top">test</Cell>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '0 0 auto');
  expect(wrapper).toHaveStyleRule('width', '100%');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
  expect(wrapper).toHaveStyleRule('align-self', 'flex-start');
});

test('cell offset', () => {
  const wrapper = shallow(<Cell smallOffset={1}>test</Cell>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('flex', '0 0 auto');
  expect(wrapper).toHaveStyleRule('width', '100%');
  expect(wrapper).toHaveStyleRule('min-height', '0px');
  expect(wrapper).toHaveStyleRule('min-width', '0px');
  expect(wrapper).toHaveStyleRule('margin-left', '8.333333333333334%');
});
