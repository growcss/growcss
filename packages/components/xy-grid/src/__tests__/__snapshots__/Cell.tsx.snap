// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell align 1`] = `
<styled.div
  align="top"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell auto layout with margin gutter 1`] = `
<styled.div
  cellType="auto"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  gutterType="margin"
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell auto layout with padding gutter 1`] = `
<styled.div
  cellType="auto"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  gutterType="padding"
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell base layout 1`] = `
<styled.div
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell offset 1`] = `
<styled.div
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  smallOffset={1}
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell shrink layout 1`] = `
<styled.div
  cellType="shrink"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</styled.div>
`;

exports[`cell with 6 small layout 1`] = `
<styled.div
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  small={6}
  vertical={false}
>
  test
</styled.div>
`;
